kernel/ramfs.c

```diff
@@ -1283,6 +1305,13 @@ static ssize_t _fs_pwrite(
 
         if (new_offset > _file_size(file))
         {
+            if (is_posix_shmfs_active_file(ramfs, file->shared->inode))
+            {
+                MYST_WLOG("Unsupported Operation: Attempt was made to write to "
```

Comment:
Question: why is this disallowed?

Reply:
From the PR description above -
> 
> 3. Resizing a posix shared memory file with writes or by truncation is unsupported if the file is serving any mmap request. We avoid supporting this because resizing a ramfs file can potentially move the underlying buffer.

The limitation was to keep the design simple. The limitation cases also fail loudly so that if in the future we encounter an app which uses this feature, it would be noticeable.